File: integrationtest/self-healing-integration-tests/src/main/java/de/dkeiss/taxishare/steps/RegistrationSteps.java
package de.dkeiss.taxishare.steps;

import com.github.javafaker.Faker;
import de.dkeiss.taxishare.pages.RegistrationPage;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

import static org.junit.Assert.assertEquals;


public class RegistrationSteps extends AbstractSteps {

    private final Faker faker = Faker.instance();

    @Given("the registration page is open")
    public void theRegistrationPageIsOpen() {
        openRelative(RegistrationPage.URL);
        expectPage(RegistrationPage.class);
    }

    @Given("the registration page v2 is open")
    public void theRegisterPageV2IsOpen() {
        openRelative(RegistrationPage.URL_V2);
        expectPage(RegistrationPage.class);
    }

    @When("the user registers")
    public void theUserRegisters() {
        RegistrationPage registrationPage = getCurrentPage();
        registrationPage.register(
                faker.name().username(),
                faker.internet().emailAddress(),
                faker.internet().password());
    }

    @Then("the registration is successful")
    public void theRegistrationIsSuccessful() {
        RegistrationPage registrationPage = getCurrentPage();
        String registrationSuccessMessage = registrationPage.getRegistrationSuccessMessage();
        assertEquals("User registered successfully!", registrationSuccessMessage);
    }

}

File: integrationtest_python/src/self_healing_integration_test/steps/registration_steps.py
"""Behave step implementations for the registration feature."""

from __future__ import annotations

from behave import given, then, when
from faker import Faker

from self_healing_integration_test.pages.registration import RegistrationPage
from self_healing_integration_test.pages.basepage import call_page_method, current_page, expect_page, open_relative

faker = Faker()


@given("the registration page is open")
def registration_page_is_open(context) -> None:
    open_relative(context, context.settings, RegistrationPage.DEFAULT_PATH)
    expect_page(context, context.store, RegistrationPage, RegistrationPage.DEFAULT_PATH)


@given("the registration page v2 is open")
def registration_page_v2_is_open(context) -> None:
    open_relative(context, context.settings, RegistrationPage.PATH_V2)
    expect_page(context, context.store, RegistrationPage, RegistrationPage.PATH_V2)


@when("the user registers")
def the_user_registers(context) -> None:
    page = current_page(context.store, RegistrationPage)
    username = faker.user_name()
    email = faker.email()
    password = faker.password()
    call_page_method(context, page, "register", username, email, password)


@then("the registration is successful")
def the_registration_is_successful(context) -> None:
    page = current_page(context.store, RegistrationPage)
    message = call_page_method(context, page, "registration_success_message")
    assert message == "User registered successfully!", message

